src/sql/src/plan/func.rs

```diff
@@ -1220,6 +1220,14 @@ lazy_static! {
                     Ok(e.call_unary(UnaryFunc::SqrtDec(s)))
                 })
             },
+            "timezone" => Scalar {
+                params!(String, Timestamp) => BinaryFunc::TimezoneTimestamp,
+                params!(String, TimestampTz) => BinaryFunc::TimezoneTimestampTz,
+                params!(String, Time) => BinaryFunc::TimezoneTime,
+                params!(Interval, Timestamp) => BinaryFunc::TimezoneIntervalTimestamp,
```

Comment:
I Couldn't figure out how to make the parser prioritize string time zones for timezone() when there is ambiguity between TEXT and INTERVAL.

Currently you have to explicitly cast a string to either text or interval for the query to succeed. PSQL just assumed it's a string if interval isn't explicitly mentioned.

Reply:
Something simple like ```SELECT TIMESTAMP '2001-01-01 00:00:00' AT TIME ZONE 'GMT'``` fails to parse in mz but not in pg, it gives ```ERROR:  Cannot call function timezone(unknown, timestamp): unable to determine which implementation to use; try providing explicit casts to match parameter types```.

I have to annotate the text with something like ```'GMT'::text``` to disambiguate. This should probably work fine when the time zone is pulled from an existing table and is annotated as a string or as an interval explicitly, but time zones are also often used as literals.

There is another issue I noticed: In pg, ```SELECT TIMESTAMP '2001-01-01 00:00:00' AT TIME ZONE '-05:00'::text```  and ```SELECT TIMESTAMP '2001-01-01 00:00:00' AT TIME ZONE '-05:00'::interval``` give different answers:
```2000-12-31 19:00:00+00``` and ```2001-01-01 05:00:00+00```. mz returns the second one for both, which is seemingly correct and what the user probably intends.

After some frustration and digging around, I figured out what was causing it: PostgreSQL is parsing it as a [POSIX offset](https://www.postgresql.org/docs/current/datetime-posix-timezone-specs.html), which considers the western hemisphere positive and are the opposite of the ISO-8601 which everyone knows and uses around the world. So '-05:00' actually corresponds to 'Etc/GMT-05' which is also [intentionally inverted](https://en.wikipedia.org/wiki/Tz_database#Area) for the same reason.

I'm not sure what to do with this, since I'm not sure if mz should include this legacy baggage. Do you know what CockroachDB does?